sdk/core/core/src/az_span.c

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_span.h>
+
+#include <_az_cfg.h>
+
+/**
+ * ASCII lower case.
+ */
+AZ_NODISCARD AZ_INLINE az_result_byte az_ascii_lower(az_result_byte const value) {
+  return 'A' <= value && value <= 'Z' ? value + ('a' - 'A') : value;
```

Comment:
'a' - 'A' could be stored in a constant rather than recalculating each time.  Compiler may be smart enough to lift that into a constant, but rather than rely on the compiler to do it for us we should.

Reply:
made the same comment 👯‍♂

Reply:
The problem is that C has no such concept as compile time constant except `enum`. So there are three options
1. `#define` doesn't improve perf.
2. `enum` I think, I can do it.
3. `static const` actually makes perf much worse :-)